python/pyspark/ml/common.py

```diff
@@ -53,24 +57,24 @@ def _new_smart_decode(obj):
 
 
 # this will call the ML version of pythonToJava()
-def _to_java_object_rdd(rdd):
+def _to_java_object_rdd(rdd: RDD) -> JavaObject:
     """Return an JavaRDD of Object by unpickling
 
     It will convert each Python object into Java object by Pickle, whenever the
     RDD is serialized in batch or not.
     """
-    rdd = rdd._reserialize(AutoBatchedSerializer(PickleSerializer()))
-    return rdd.ctx._jvm.org.apache.spark.ml.python.MLSerDe.pythonToJava(rdd._jrdd, True)
+    rdd = rdd._reserialize(AutoBatchedSerializer(PickleSerializer()))  # type: ignore[attr-defined]
+    return rdd.ctx._jvm.org.apache.spark.ml.python.MLSerDe.pythonToJava(rdd._jrdd, True)  # type: ignore[attr-defined]
```

Comment:
I don't get exactly why mypy doesn't recognize `_reserialize` and `_jvm` as attributes. In the case of `_reserialize`, is it the leading underscore making it a "private" method? In the case of `_jvm`, is it that it's a class attribute?

Reply:
At the moment, both `SparkContext` and `RDD` don't have inline annotations and stubs, intentionally, cover only public API.

Reply:
We didn't annotate types of `_reserialize` or`_jvm`  in https://github.com/apache/spark/blob/master/python/pyspark/rdd.pyi or https://github.com/apache/spark/blob/master/python/pyspark/context.pyi.

Reply:
I think it is fine to leave `type: ignore[attr-defined]` here. We may come back to these after inlining type hints for SparkContext for example.

Reply:
FYI https://github.com/apache/spark/pull/34466